src/main/java/core/basesyntax/handler/impl/PurchaseTransactionHandler.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.handler.impl;
+
+import core.basesyntax.handler.TransactionHandler;
+import java.util.Map;
+
+public class PurchaseTransactionHandler implements TransactionHandler {
+    @Override
+    public void handleTransaction(Map<String, Integer> mapFruitQuantity, String key, int value) {
+        mapFruitQuantity.put(key, mapFruitQuantity.getOrDefault(key, 0) - value);
```

Comment:
When handling a Purchase operation, you must ensure that the resulting balance of the fruit is not negative. If the balance would go negative, you should throw a RuntimeException to prevent invalid transactions. For example, you can check if `mapFruitQuantity.getOrDefault(key, 0) >= value` before subtracting.

Reply:
When handling a Purchase operation, you must ensure that the resulting balance of the fruit is not negative